pkgs/top-level/darwin-packages.nix

```
@@ -223,14 +223,15 @@ impure-cmds // appleSourcePackages // chooseLibs // {
   discrete-scroll = callPackage ../os-specific/darwin/discrete-scroll { };
 
```

I think we should have a cross compiled version of the linux builder available. Gabriella's solution is great and robust / low maintenance because it reuses existing builds. However, as an alternate, potentially simpler solution to the bootstrapping problem, it would be great to have a cross compiled configuration on Hydra and in the cache (although the cache is not a requirement in this case, which is actually the whole point, it's always nice to have some cache).


```suggestion
  linuxBuilder_cross = linuxBuilder.override (o: { modules = o.modules ++ [{
    nixpkgs.buildPlatform = stdenv.buildPlatform;
  }]; });
```

I think you'd also have to change `nixpkgs.system` in the default config to `nixpkgs.hostPlatform`.

(These `*Platform` ones are new options that are a bit friendlier than the old `crossSystem`/`localSystem` combination.)

---

I'm not too familiar with cross building, happy to investigate in another PR if that's okay with you.

---

Of course; just wanted to put the idea out there.